.github/workflows/main.yml

```diff
@@ -0,0 +1,155 @@
+name: Main CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        version: [max] # [x.x.x | latest | max]
+        type: [stable] # [stable | insider]
+      fail-fast: false
+
+    env:
+      CODE_VERSION: ${{ matrix.version }}
+      CODE_TYPE: ${{ matrix.type }}
+      VSCODE_CAMELK_GITHUB_TOKEN: ${{ github.token }}
+
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+          cache: npm
+
+      - name: Setup vsce
+        run: |
+          npm install -g @vscode/vsce
+          vsce --version
+
+      - name: Prepare macos runner
+        if: matrix.os == 'macos-latest'
+        run: |
+          brew install -f docker colima kubectl
+          colima start
+          docker ps
+
+      - name: Start minikube
+        uses: medyagh/setup-minikube@v0.0.11
+        with:
+          # TODO: from versin 1.29.0 macos runner is broken:
+          #       Error: The process '/Users/runner/bin/minikube' failed with exit code 109
+          minikube-version: 1.28.0
+          driver: docker
+          addons: registry
+
+      - name: Configure Kamel (Ubuntu)
+        if: matrix.os == 'ubuntu-latest'
+        run: |
+          # Download and provide in path kamel.
```

Comment:
same as comment on insider workflow

if I see correctly the only difference between linux/macos Configure Kamel step is the .tar.gz naming and other lines of code are same.

could you please to extract download kamel bin archive step which could be parametrised or solved by if statement and next would be the Configure Kamel step which is general for both systems? is that possible?